static/js/stream_data.js

```diff
@@ -556,6 +557,13 @@ export function can_subscribe_others(sub) {
     return !page_params.is_guest && (!sub.invite_only || sub.subscribed);
 }
 
+export function can_unsubscribe_others(sub) {
+    return user_groups.is_user_in_group(
+        sub.can_remove_subscribers_group_id,
+        people.my_current_user_id(),
+    );
```

Comment:
Thinking about this behavior, I think we probably are going to want to still allow organization administrators to do this management as they could before this setting. (Later, we'll probably have an organization-level setting defining a group too, and you'll be able to do the thing if either you have the global permission or the local one for this stream).

Reply:
Do we want to keep the owner option for now in the dropdown if we are anyways allowing admin to unsubscribe the user? 

Also, would like to mention that admins can still change the setting, so eventually they can just change the setting if it is set to "owners only". We might also want to think about how to handle this.

Reply:
Hm, that sounds like we should drop the org admins option.

Reply:
Yes, sounds good. Will update the PR.

Reply:
I added some long comments here as well as a couple checks to make this match what the server is doing.

I noted one minor bug in this comment that I think we shouldn't worry about, but there's also a bug I think we need to fix: I think the "Unsubscribe" buttons in the "User profile" modals likely don't handle this right. I think this is roughly the fix (and seeing this diff encourages me to do the refactoring suggested in the comments I add):

```
diff --git a/static/js/user_profile.js b/static/js/user_profile.js
index bc4e7658dd..411eb82d43 100644
--- a/static/js/user_profile.js
+++ b/static/js/user_profile.js
@@ -55,7 +55,7 @@ function initialize_bot_owner(element_id, bot_id) {
 
 function format_user_stream_list_item(stream, user) {
     const show_unsubscribe_button =
-        people.can_admin_user(user) || settings_data.user_can_unsubscribe_other_users();
+        people.can_admin_user(user) || settings_data.user_can_unsubscribe_other_users() || stream_data.can_unsubscribe_others(stream);
     const show_private_stream_unsub_tooltip =
         people.is_my_user_id(user.user_id) && stream.invite_only;
     return render_user_stream_list_item({
```

This looks a bit ugly though, with the rows having buttons being taller than the others:
![image](https://user-images.githubusercontent.com/2746074/216842743-a1819802-1f0d-4e9b-8e9c-d8995bbdcc55.png). Probably the fix is a lower height on the button?

I ended up adding a couple commits to address this in a way that seems OK, with more cleanup.